chalk-solve/src/display/state.rs

```diff
@@ -109,12 +216,22 @@ impl<'a, I: Interner> WriterState<'a, I> {
         std::iter::repeat("  ").take(self.indent_level).format("")
     }
 
-    pub(super) fn alias_for_id_name(&self, id: I::DefId, name: String) -> impl Display {
-        self.def_id_aliases.borrow_mut().alias_for_id_name(id, name)
+    pub(super) fn alias_for_adt_id_name(&self, id: I::InternedAdtId, name: String) -> impl Display {
```

Comment:
I wonder if these could be simplified by making this `id: impl Into<UnifiedId<I>>`.

Reply:
We tried to, but I think using `Into<UnifiedId<I>>` in general has some significant issues. Specifically, we would need to have two implementations

```rust
impl<I: Interner> From<I::DefId> for UnifiedId<I> { ... }

impl<I: Interner> From<I::InternedAdtId> for UnifiedId<I> { ... }
```

and these end up conflicting, since there's no guarantee that `DefId` and `InternedAdtId` are actually different types for an arbitrary interner.